File: 01_BasicPagramming/Lab01/Python_Lang/Demo16.py
#
#  Created by NDHuy at 16/04/2020
#  Copyright 2020 by SEstudio
#  Find Fibonacci n with:
#      a0 = a1 = 1
#      a2 = a1 + a0
#      ...
#      an = a(n - 1) + a(n - 2)
#

def main():
    n = int(input("Enter a integer n = "))

    if (n < 0):
        print("Cannot find Fibonacci for a(%d)" %(n))
    else:
        if (n == 0) or (n == 1):
            a_n = 1
        else:
            a_1 = 1
            a_2 = 1
            i = 2
            while (i <= n):
                a_n = a_1 + a_2
                a_2 = a_1
                a_1 = a_n
                i += 1
        print("Fibonacci a(%d) = %d" %(n, a_n))


if __name__ == "__main__":
    main()

File: Session_01_BasicPagramming/Lab01/C_Lang/src/Demo16.cpp
//
//  Created by NDHuy at 16/04/2020
//  Copyright 2020 by SEstudio
//  Find Fibonacci n with:
//      a0 = a1 = 1
//      a2 = a1 + a0
//      ...
//      an = a(n - 1) + a(n - 2)
//

#include <stdio.h>

int main()
{
    int n; 
    printf("Enter a integer n = ");
    scanf("%d", &n);

    unsigned long a_n;    // a_n = a(n)
    unsigned long a_1;    // a_1 = a(n - 1)
    unsigned long a_2;    // a_2 = a(n - 2)

    if(n < 0)
    {
        printf("Cannot find Fibonacci for a(%d) \n", n);
    }
    else
    {
        // Expecial case (STOP Case)
        if ((n == 0) || (n == 1))
            a_n = 1;
        else
        {
            a_1 = 1;    // a0 = 1
            a_2 = 1;    // a1 = 1
            for(int i = 2; i <= n; i++)
            {
                a_n = a_1 + a_2;
                a_2 = a_1;
                a_1 = a_n;
            }
        }
        
        printf("Fibonacci a(%d) = %ld \n", n, a_n);
    }
    
    return 0;
}